client/rpc-core/debug/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use ethereum_types::H256;
+use jsonrpc_core::Result;
+use jsonrpc_derive::rpc;
+use serde::Deserialize;
+
+pub use crate::types::{StepLog, TraceExecutorResponse};
+
+pub use rpc_impl_Debug::gen_server::Debug as DebugServer;
+
+pub mod types {
+	pub use moonbeam_rpc_primitives_debug::{StepLog, TraceExecutorResponse};
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TraceParams {
+	/// Javascript tracer (we just check if it's Blockscout tracer string)
+	pub tracer: Option<String>,
+}
+
+#[rpc(server)]
+pub trait Debug {
+	#[rpc(name = "debug_traceTransaction")]
+	fn trace_transaction(
+		&self,
+		transaction_hash: H256,
+		params: Option<TraceParams>,
+	) -> Result<TraceExecutorResponse>;
```

Comment:
Why are there two `Option`s in here? Is there a use for `params = Some(TraceParams{tracer: None})`?

Reply:
`TraceParams` parameter include more than the tracer field (`disableStorage`, `disableMemory`, `disableStack`) - but they are currently unimplemented.

Just added the fields, a TODO comment and opened MOON-420.